lib/DocumentManagerContext.php

```diff
@@ -0,0 +1,199 @@
+<?php
+
+/*
+ * This file is part of Sulu.
+ *
+ * (c) MASSIVE ART WebServices GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Component\DocumentManager;
+
+use PHPCR\SessionInterface;
+use ProxyManager\Factory\LazyLoadingGhostFactory;
+use Symfony\Component\EventDispatcher\EventDispatcherInterface;
+
+/**
+ * The document manager context is a container for all the dependencies of a
+ * given document manager.
+ */
+class DocumentManagerContext
+{
+    /**
+     * @var EventDispatcherInterface
+     */
+    private $eventDispatcher;
+
+    /**
+     * @var DocumentInspectorFactory
+     */
+    private $inspectorFactory;
+
+    /**
+     * @var NodeManager
+     */
+    private $nodeManager;
+
+    /**
+     * @var DocumentRegistry
+     */
+    private $registry;
+
+    /**
+     * @var ProxyFactory
+     */
+    private $proxyFactory;
+
+    /**
+     * @var MetadataFactoryInterface
+     */
+    private $metadataFactory;
+
+    /**
+     * @var SessionInterface
+     */
+    private $session;
+
+    /**
+     * @var DocumentManagerInterface
+     */
+    private $manager;
+
+    /**
+     * @var string
+     */
+    private $name;
+
+    public function __construct(
+        $name,
+        SessionInterface $session,
+        EventDispatcherInterface $eventDispatcher,
+        MetadataFactoryInterface $metadataFactory,
+        LazyLoadingGhostFactory $lazyProxyFactory,
+        DocumentInspectorFactoryInterface $inspectorFactory,
+        $defaultLocale
+    ) {
+        $this->name = $name;
+
+        // the event dispatcher should be unique to this document manager context
+        // instance.
+        $this->eventDispatcher = $eventDispatcher;
+
+        // the inspector factory provides a way for users to instantiate
+        // their own document inspectors.
+        $this->inspectorFactory = $inspectorFactory;
+
+        // the metadata factory is currently intended to be the same for all
+        // document mnaager contexts instances.
+        $this->metadataFactory = $metadataFactory;
+
+        // the PHPCR session SHOULD be unique to all (end user) document
+        // manager contexts.
+        $this->session = $session;
+
+        // instantiate other objects scoped to this document manager.
+        $this->nodeManager = new NodeManager($session);
+        $this->proxyFactory = new ProxyFactory($this, $lazyProxyFactory, $metadataFactory);
+        $this->registry = new DocumentRegistry($defaultLocale);
+        $this->manager = new DocumentManager($this);
```

Comment:
I don't really get the separation of creating new instances vs. injecting something. I would simply inject all the services, so that they are also available from the container if necessary. Or if we are consequent the other way round we should also create the `EventDispatcher` like that. But I would like to decide that together (or didn't we already say that we export that kind of stuff to the Extension?)